pkgs/by-name/qu/qubes-seabios/package.nix

```diff
@@ -0,0 +1,110 @@
+{
+  lib,
+  stdenv,
+  stdenvNoCC,
+  fetchFromGitHub,
+  python3,
+  seabios,
+}:
+let
+  inherit (lib) concatStringsSep map;
+
+  version = "4.2.0";
+  src = fetchFromGitHub {
+    owner = "QubesOS";
+    repo = "qubes-seabios";
+    rev = "refs/tags/R${version}";
+    hash = "sha256-KEvdQ0VLJSnLRNuzMBWY0LFp9CN1EKq3K2v8Ud/vojc=";
+  };
+  mkBios =
+    {
+      config,
+      target ? "",
+      outBin,
+      name,
+    }:
+    stdenv.mkDerivation {
+      inherit version;
+      src = seabios.src;
```

Comment:
These `src` calls seem weird. the `seabios.src` isn't used at all. Is this right?

Reply:
It is used, what's wrong? mkBios function is called multiple times

Reply:
Since the `let..in` statement is recursive, it isn't actually used. `mkBios` uses the `src` attribute defined earlier, while `seabios.src` is unused.

Reply:
?..

There is explicit `src = seabios.src` line, and no `inherit src`, which will not work since this will cause duplicate attribute error.

It also does work, after all :D

Reply:
src is only used at `cp ${src}/${config} .config` line, and nowhere else

Reply:
`nixd` tells me there's an error with that `cp` line if I remove the Qubes `src`, while no error occurs if I remove the `seabios.src` `src`. This leads me to believe the latter is unused.

Either that, or we've discovered a bug in `nixd`!

Reply:
Fist src is an argument of mkDerivation, which will fill $src env var for unpackPhase & others, nixd has no idea if this enviroment variable is used or not, so it can't report an error.

If you remove `src = seabios.src`, then you will get package build failure.

Reply:
Default unpackPhase works by extracting $src into working directory, then by cdIng into sourceRoot
And then it does patchPhase, configurePhase, buildPhase, installPhase

Reply:
Alright, that makes sense! Sorry for the confusion.

Reply:
To avoid confusion, qubes src might be renamed to patches instead